pkg/ebpf/c/tracee.bpf.c

```diff
@@ -3486,27 +3529,71 @@ int tracepoint__sched__sched_process_exec(struct bpf_raw_tracepoint_args *ctx)
     // We passed all filters (in should_trace()) - add this pid to traced pids set
     data.task_info->follow = true;
 
-    struct task_struct *task = (struct task_struct *) ctx->args[0];
-    struct linux_binprm *bprm = (struct linux_binprm *) ctx->args[2];
+    if (should_submit(SCHED_PROCESS_EXEC, data.config) || data.config->options & OPT_PROCESS_INFO) {
+        // Note: Starting from kernel 5.9, there are two new interesting fields in bprm that we
+        // should consider adding:
+        // 1. struct file *executable - can be used to get the executable name passed to an
+        // interpreter
+        // 2. fdpath                  - generated filename for execveat (after resolving dirfd)
+        struct linux_binprm *bprm = (struct linux_binprm *) ctx->args[2];
+
+        if (bprm == NULL) {
+            return -1;
+        }
 
-    if (bprm == NULL) {
-        return -1;
-    }
+        struct file *file = get_file_ptr_from_bprm(bprm);
+        void *file_path = get_path_str(GET_FIELD_ADDR(file->f_path));
+        const char *filename = get_binprm_filename(bprm);
 
-    int invoked_from_kernel = 0;
-    if (get_task_parent_flags(task) & PF_KTHREAD) {
-        invoked_from_kernel = 1;
+        dev_t s_dev = get_dev_from_file(file);
+        unsigned long inode_nr = get_inode_nr_from_file(file);
+        u64 ctime = get_ctime_nanosec_from_file(file);
+        umode_t inode_mode = get_inode_mode_from_file(file);
+
+        // The map of the interpreter will be updated for any loading of an elf, both for the elf
+        // and for the interpreter. Because the interpreter is loaded only after the executed elf is
+        // loaded, the map value of the executed elf should be overridden by the interpreter.
+        file_info_t *elf_interpreter =
+            bpf_map_lookup_elem(&interpreter_map, &data.context.task.host_tid);
+
+        save_str_to_buf(&data, (void *) filename, 0);
+        save_str_to_buf(&data, file_path, 1);
+        save_to_submit_buf(&data, &s_dev, sizeof(dev_t), 4);
+        save_to_submit_buf(&data, &inode_nr, sizeof(unsigned long), 5);
+        save_to_submit_buf(&data, &ctime, sizeof(u64), 7);
```

Comment:
I see that some of the fields are not ordered according to the event's parameters.
I think we should either change the order in events.go as well to fit what we have here, or split this program and the tailcall so the indexes and order will be kept as they were.